docs/common_links.rst

```diff
@@ -5,40 +5,41 @@
+.. |DimensionlessParticle| replace:: :py:class:`~plasmapy.particles.particle_class.DimensionlessParticle`
+.. |IonicLevel| replace:: :py:class:`~plasmapy.particles.ionization_state.IonicLevel`
+.. |IonizationState| replace:: :py:class:`~plasmapy.particles.ionization_state.IonizationState`
+.. |IonizationStateCollection| replace:: :py:class:`~plasmapy.particles.ionization_state_collection.IonizationStateCollection`
+.. |Particle| replace:: :py:class:`~plasmapy.particles.particle_class.Particle`
+.. |particle_input| replace:: :py:decorator:`~plasmapy.particles.decorators.particle_input`
+.. |ParticleLike| replace:: :py:obj:`~plasmapy.particles.particle_class.ParticleLike`
+.. |ParticleList| replace:: :py:class:`~plasmapy.particles.particle_collections.ParticleList`
 
-.. |ChargeError| replace:: :class:`~plasmapy.particles.exceptions.ChargeError`
-.. |InvalidElementError| replace:: :class:`~plasmapy.particles.exceptions.InvalidElementError`
-.. |InvalidIonError| replace:: :class:`~plasmapy.particles.exceptions.InvalidIonError`
-.. |InvalidIsotopeError| replace:: :class:`~plasmapy.particles.exceptions.InvalidIsotopeError`
-.. |InvalidParticleError| replace:: :class:`~plasmapy.particles.exceptions.InvalidParticleError`
-.. |MissingParticleDataError| replace:: :class:`~plasmapy.particles.exceptions.MissingParticleDataError`
-.. |MissingParticleDataWarning| replace:: :class:`~plasmapy.particles.exceptions.MissingParticleDataWarning`
-.. |ParticleError| replace:: :class:`~plasmapy.particles.exceptions.ParticleError`
-.. |ParticleWarning| replace:: :class:`~plasmapy.particles.exceptions.ParticleWarning`
-.. |UnexpectedParticleError| replace:: :class:`~plasmapy.particles.exceptions.UnexpectedParticleError`
+.. |ChargeError| replace:: :py:exception:`~plasmapy.particles.exceptions.ChargeError`
+.. |InvalidElementError| replace:: :py:exception:`~plasmapy.particles.exceptions.InvalidElementError`
+.. |InvalidIonError| replace:: :py:exception:`~plasmapy.particles.exceptions.InvalidIonError`
+.. |InvalidIsotopeError| replace:: :py:exception:`~plasmapy.particles.exceptions.InvalidIsotopeError`
+.. |InvalidParticleError| replace:: :py:exception:`~plasmapy.particles.exceptions.InvalidParticleError`
+.. |MissingParticleDataError| replace:: :py:exception:`~plasmapy.particles.exceptions.MissingParticleDataError`
+.. |MissingParticleDataWarning| replace:: :py:exception:`~plasmapy.particles.exceptions.MissingParticleDataWarning`
+.. |ParticleError| replace:: :py:exception:`~plasmapy.particles.exceptions.ParticleError`
+.. |ParticleWarning| replace:: :py:exception:`~plasmapy.particles.exceptions.ParticleWarning`
+.. |UnexpectedParticleError| replace:: :py:exception:`~plasmapy.particles.exceptions.UnexpectedParticleError`
 
 .. ------------------
 .. NumPy replacements
 .. ------------------
 
-.. |ndarray| replace:: :class:`numpy.ndarray`
+.. |ndarray| replace:: :py:class:`numpy.ndarray`
```

Comment:
Note, you do not have to include `:py:` here because it is the default domain for sphinx.  It is assumed unless a different domain is given.

https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-primary_domain